Author: Pino Toscano <pino@kde.org>
Description: fix GooString::insert
Applied-Upstream: 0.21.3, commit:7ba15d11e56175601104d125d5e4a47619c224bf, commit:55940e989701eb9118015e30f4f48eb654fa34c4
Last-Update: 2012-11-27
Bug-Debian: http://bugs.debian.org/693817

--- a/goo/GooString.cc
+++ b/goo/GooString.cc
@@ -640,14 +640,12 @@ GooString *GooString::insert(int i, GooS
 }
 
 GooString *GooString::insert(int i, const char *str, int lengthA) {
-  int j;
   int prevLen = length;
   if (CALC_STRING_LEN == lengthA)
     lengthA = strlen(str);
 
   resize(length + lengthA);
-  for (j = prevLen; j >= i; --j)
-    s[j+lengthA] = s[j];
+  memmove(s+i+lengthA, s+i, prevLen-i);
   memcpy(s+i, str, lengthA);
   return this;
 }
